### PR TITLE
Updated auth docs, fixed broken UI Component example

### PR DESCRIPTION
**Pull request by dabit3**
Fixed broken React UI getting started example (usage.md), improved Auth docs for JavaScript.

### CODE REVIEW

**Comment by dabit3** [comment]
Added additional changes to update Auth docs.

File: docs/lib/auth/fragments/js/emailpassword.md [hunk 1]
```diff
@@ -1,5 +1,109 @@
...
+try {
+    await Auth.signOut({ global: true });
+} catch (error) {
+    console.log('error signing out: ', error);
+}
+```
+
+## Advanced use cases
+
+### Sign-in
```
Comment (swaminator) [inline docs/lib/auth/fragments/js/emailpassword.md]: Can we address exactly what usecase this addresses? E.g. Additional validations at sign-in

Comment (dabit3) [reply docs/lib/auth/fragments/js/emailpassword.md]: Thanks, just updated with additional info in headings for tthe two cases covered here.

File: docs/lib/auth/fragments/js/emailpassword.md [hunk 2]
```diff
@@ -1,5 +1,117 @@
...
+import { Auth } from 'aws-amplify';
+
+try {
+    await Auth.signOut({ global: true });
+} catch (error) {
+    console.log('error signing out: ', error);
+}
+```
+
+### Forcing Email Uniqueness in Cognito User Pools
```
Comment (swaminator) [inline docs/lib/auth/fragments/js/emailpassword.md]: one last nit: Can we add this in the `Advanced use cases` section?

Comment (dabit3) [reply docs/lib/auth/fragments/js/emailpassword.md]: Ok, just pushed an update with this.

**Pull request merged**